mkt/websites/tests/test_views.py

```diff
@@ -244,3 +246,114 @@ def test_status_filtering(self):
+        responses.add(responses.GET, URL, status=500)
+        response, content = self.go(url=URL)
+        eq_(response.status_code, 400)
+        eq_(content, WebsiteMetadataScraperView.errors['network'])
+
+    @responses.activate
+    def test_empty_body(self):
+        URL = 'https://marketplace.firefox.com/'
+        responses.add(responses.GET, URL, '', status=200)
+        response, content = self.go(url=URL)
+        eq_(response.status_code, 400)
+        eq_(content, WebsiteMetadataScraperView.errors['malformed_data'])
+
+    @responses.activate
+    def test_valid(self):
+        URL = 'https://marketplace.firefox.com/'
+        responses.add(responses.GET, URL, '<html />', status=200)
+        response, content = self.go(url=URL)
+        eq_(response.status_code, 200)
+
+    def test_verbs(self):
+        self._allowed_verbs(self.url, ['get'])
+
+    def test_has_cors(self):
+        self.assertCORS(self.client.get(self.url), 'get')
+
+
+class TestWebsiteScrapeName(TestCase):
+    """
+    WebsiteMetadataScraperView.get_name will attempt to fetch an appropriate
+    name for a passed PyQuery document. Here, we're unit testing that logic.
+
+    get_name searches for an appropriate website name using a preference order
+    that reverses the order of self.fragments. We test that it works by:
+
+    1) Constructing an HTML document with no children, and ensuring that
+       get_name returns None.
+    2) Constructing a document with only <meta name="application-name" />, and
+       ensuring that get_name returns the value of that tag's `content`
+       attribute.
+    3) Constructing a document with both <meta name="application-name" /> and
+       <meta name="apple-mobile-web-app-title" />, ensuring that get_name
+       returns the value of the latter's `content` attribute.
+    4) And so on, adding an item from self.fragments for each subsequent
+       iteration.
+    """
+    fragments = [
+        '<meta name="application-name" content="%s" />',
+        '<meta name="apple-mobile-web-app-title" content="%s" />',
+        '<meta property="og:title" content="%s" />',
+        '<title>%s</title>'
+    ]
+    value = 'Pied Piper %s'
+
+    def setUp(self):
+        self.view = WebsiteMetadataScraperView()
+        super(TestWebsiteScrapeName, self).setUp()
+
+    def test_no_values(self):
+        """
+        Test that None is returned when we can't detect any metadata.
+        """
+        doc = pq('<html></html>')
+        eq_(None, self.view.get_name(doc))
+
+    def get_value(self, iteration):
+        return self.value % str(iteration)
+
+    def get_fragment(self, iteration):
+        return self.fragments[iteration - 1] % self.get_value(iteration)
+
+    def get_document(self, iteration):
+        doc = '<html>%s</html>'
+        inner = [self.get_fragment(i + 1) for i in xrange(0, iteration)]
+        return pq(doc % ''.join(inner))
+
+    def test_values(self):
+        for n in xrange(0, len(self.fragments)):
```

Comment:
I find the adding and subtracting one pretty confusing. I understand what's supposed to be happening but it is quite distracting. It would be nice to pull the complexity into a generator or update the loop to start at 1 (or both).

```py
for document, result in self.get_test_data():
    eq_(result, self.view.get_name(document))